Sources/AsyncHTTPClient/RequestBag.swift

```diff
@@ -168,11 +171,13 @@ final class RequestBag<Delegate: HTTPClientResponseDelegate> {
         switch action {
         case .none:
             break
-        case .forwardStreamFinished(let writer, let promise):
-            writer.finishRequestBodyStream(self)
-            promise?.succeed(())
-
-            self.delegate.didSendRequest(task: self.task)
+        case .forwardStreamFinished(let writer, let writerPromise):
+            let promise = self.task.eventLoop.makePromise(of: Void.self)
+            promise.futureResult.whenSuccess {
+                self.delegate.didSendRequest(task: self.task)
+            }
+            writer.finishRequestBodyStream(self, promise: promise)
+            writerPromise?.succeed(())
```

Comment:
Why do we create a new promise? Why is the writePromise just succeeded? This looks odd. This needs at least a comment.

Reply:
I thought that we need a new promise here but I just tested it again and with using the same promise and it worked.